### DISCUSSION TITLE
Troubles Integrating diesel_migration in my library

### DISCUSSION THREAD

**Discussion by case-prudolicce** [diesel-rs/diesel#3003] [Q&A]
Hello, I am currently stuck with one of my projects, [IntelCache](https://github.com/case-prudolicce/IntelCache), with inplementing an init command for my server.

The issue I have is implementing `diesel_migrations` in the [library backend](https://github.com/case-prudolicce/IntelCache/blob/main/intel-cache-lib/src/lib_backend.rs) portion of the code (which would be called by servers and any rust based clients). The idea is that [my server](https://github.com/case-prudolicce/IntelCache/blob/main/intel-cache-server/src/main.rs) would call a function from that library (init,build_server, whatever)

Thing is, I've tried doing that. And at first glance it works but when verifying mysql it clearly isn't doing anything. However there aren't any errors or warnings to point me in any direction. 

Im wondering why this isn't working and how/what can I do about this.

I dont have the code but if I recall added a `build_sql` function in libbackend. Libbackend had the `embed_migrations!` with the right relative path,`#[macro_use]` on top of `extern crate diesel_migrations`, the `build_sql` function called `embedded_migrations::run(&db_con)` and the`db_con` was verified. `build_sql` was being called from another crate (server) within the same workspace. No errors, no warnings, seems to work, empty databse. I'm at a loss of words.

**Comment by weiznich** [comment]
I'm not sure what information you like to get here, as this strategy is exactly the right way to handle migrations using the `embed_migrations!` macro.
